src/components/Common/molecules/SearchFilter/index.tsx

```diff
@@ -35,9 +35,15 @@ const SearchFilter = ({ filterType, onSubmit }: SearchFilterTypeProps) => {
       : value === '여자'
       ? (value = 'WOMAN')
       : '';
-    const copy = [...filterState];
-    copy[idx] = value;
-    setFilterState(copy);
+    if (filterState[idx] === value) {
+      const copy = [...filterState];
+      copy[idx] = undefined;
+      setFilterState(copy);
+    } else {
+      const copy = [...filterState];
+      copy[idx] = value;
+      setFilterState(copy);
+    }
```

Comment:
```suggestion
    const copy = [...filterState];
    copy[idx] = filterState[idx] === value ? undefined : value
    setFilterState(copy);  
```
이렇게 리팩토링 해도 될 것 같은데 어떤가요?

Reply:
좋네요